### ISSUE TITLE
Neovim crashes if pressing the toggle command in fzf inside toggleterm

### ISSUE THREAD

**Issue by zakissimo** [neovim/neovim#17956]
### Neovim version (nvim -v)

NVIM v0.7.0-dev+1275-g00effff56

### Vim (not Nvim) behaves the same?

no

### Operating system/version

Linux desk 5.17.1-arch1-1 #1 SMP PREEMPT Mon, 28 Mar 2022 20:55:33 +0000 x86_64 GNU/Linux

### Terminal name/version

kitty 0.24.4

### $TERM environment variable

kitty

### Installation

AUR

### How to reproduce the issue

Install [toggleterm](https://github.com/akinsho/toggleterm.nvim/) plugin
Install [fzf](https://github.com/junegunn/fzf)
Open neovim
Open toggleterm
Type fzf
Press command to close toggleterm

### Expected behavior

I expect toggleterm to close and not crash neovim

### Actual behavior

![togggleterm_bug](https://user-images.githubusercontent.com/64891486/161290295-05a31fc4-e69a-4a91-b248-9eacc8dbe409.gif)

My toggle command is ```<C-c>```.
As you can see, the first time I pressed ```<C-c>```, it successfully toggled the terminal off, but the second time, neovim crashed.

Here's my config:

```lua
local status_ok, toggleterm = pcall(require, "toggleterm")
if not status_ok then
	return
end

local opt = { noremap = true }
local map = vim.api.nvim_set_keymap
-- local opts = { noremap = true, silent = true }

toggleterm.setup({
	size = 20,
	open_mapping = [[<C-c>]],
	hide_numbers = true,
	shade_filetypes = {},
	shade_terminals = true,
	shading_factor = 2,
	start_in_insert = true,
	insert_mappings = true,
	terminal_mappings = true,
	persist_size = true,
	direction = "float",
	close_on_exit = true,
	shell = vim.o.shell,
	float_opts = {
		border = "curved",
		winblend = 0,
		highlights = {
			border = "Normal",
			background = "Normal",
		},
	},
})

function _G.set_terminal_keymaps()
	vim.api.nvim_buf_set_keymap(0, "t", "<C-q>", [[<C-\><C-n>]], opt)
	-- vim.api.nvim_buf_set_keymap(0, "t", "<C-h>", [[<C-\><C-n><C-W>h]], opt)
	-- vim.api.nvim_buf_set_keymap(0, "t", "<C-j>", [[<C-\><C-n><C-W>j]], opt)
	-- vim.api.nvim_buf_set_keymap(0, "t", "<C-k>", [[<C-\><C-n><C-W>k]], opt)
	-- vim.api.nvim_buf_set_keymap(0, "t", "<C-l>", [[<C-\><C-n><C-W>l]], opt)
end

vim.cmd("autocmd! TermOpen term://* lua set_terminal_keymaps()")

local Terminal = require("toggleterm.terminal").Terminal
local lazygit = Terminal:new({ cmd = "lazygit", hidden = true })

function _LAZYGIT_TOGGLE()
	lazygit:toggle()
end

-- map("t", "<esc>", "<C-\\><C-N>", opts)
map("n", "<Leader>gg", ":lua _LAZYGIT_TOGGLE()<CR>", opt)
```

I tried changing my toggle command to ```<C-t>``` and the bug is still there.
I can't reproduce on normal terminal.

Reported to toggleterm owner [here](https://github.com/akinsho/toggleterm.nvim/issues/189).
Following his suggestion to report the bug.

**Comment by seandewar** [comment]
Maybe #16699. Do you have a backtrace?

**Comment by zakissimo** [comment]
I don't really know how  to get a backtrace, can you point me towards the proper documentation ?

**Comment by dundargoc** [comment]
https://github.com/neovim/neovim/wiki/FAQ#backtrace-linux

**Comment by zakissimo** [comment]
Thank you, here's what I have (I reproduced the bug successfully right before generating the backtrace file), I hope it's what you need:
[bt.txt](https://github.com/neovim/neovim/files/8399596/bt.txt)

**Comment by zeertzjq** [comment]
Can you try to see it this solves the problem:
```
sudo pacman -Sdd --asdeps libvterm
```

**Comment by windwp** [comment]
+1 i have same problem on  nvim 0.7 It happen randomly and i don't know how to reproduce but it happen after i quit a neovim terminal.

**Comment by zakissimo** [comment]
> Can you try to see if this solves the problem:
> 
> ```
> sudo pacman -Sdd --asdeps libvterm
> ```
> 
> and replace `libvterm-0.1` with `libvterm`.
> 
> You will still need `libvterm-0-1` for building Nvim though.

This is what I get when trying what you suggest:

```bash
~ ✖ sudo pacman -Sdd --asdeps libvterm
warning: libvterm-0.2-1 is up to date -- reinstalling
looking for conflicting packages...

Packages (1) libvterm-0.2-1

Total Installed Size:  0.13 MiB
Net Upgrade Size:      0.00 MiB

:: Proceed with installation? [Y/n] 
```

I don't understand how I should replace libvterm-0.1 with libvterm.

**Comment by zeertzjq** [comment]
Oh then this is unrelated.

**Comment by zeertzjq** [comment]
Oh, are you are using `neovim-nightly-bin`?

**Comment by zakissimo** [comment]
Yes I'm on nightly. Should I try to reproduce with stable ?

**Comment by zeertzjq** [comment]
No, I did not mean that.

**Comment by zeertzjq** [comment]
You can try to seem if this solves the problem then:
1. Install `neovim-git` from AUR (this builds Nvim from source and installs `libvterm-0.1`)
2. Run `sudo pacman -Sdd --asdeps libvterm` and replace `libvterm-0.1` with `libvterm`

**Comment by zakissimo** [comment]
> You can try to seem if this solves the problem then:
> 
> 1. Install `neovim-git` from AUR (this builds Nvim from source and installs `libvterm-0.1`)
> 2. Run `sudo pacman -Sdd --asdeps libvterm` and replace `libvterm-0.1` with `libvterm`

Yes it fixed the issue ! I keep toggling and it doesn't crashes anymore ! Thanks, legend!

**Comment by zeertzjq** [comment]
You can use this as the workaround until a libvterm 0.2 version bump PR is merged (#17329). Right now #17329 is blocked by a libvterm bug. Since libvterm 0.2 fixes this issue, this is a duplicate of #16699.

**Comment by neovim-discourse** [comment]
This issue has been mentioned on **Neovim Discourse**. There might be relevant details there:

https://neovim.discourse.group/t/how-to-debug-intermittent-neovim-freezes/2225/4
